pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -787,19 +794,27 @@ func (nc *Controller) monitorNodeHealth() error {
 				nc.processNoTaintBaseEviction(node, &observedReadyCondition, gracePeriod)
 			}
 
-			// Report node event.
-			if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
+			_, needsRetry := nc.nodesToRetry.Load(node.Name)
+			// We call markPodsNotReadyFunc when node condition changed from true to something different
+			// or previous markPodsNotReadyFunc failed and ready status is still not true.
+			if (currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue) ||
+				(needsRetry && observedReadyCondition.Status != v1.ConditionTrue) {
+				// Report node event.
```

Comment:
This isn't what we want - the fact that we didn't manager to mark pods as not-ready doesn't mean we want to report the node event again.